src/ParseObject.js

```diff
@@ -1046,7 +1093,11 @@ export default class ParseObject {
   /** Static methods **/
 
   static _clearAllState() {
-    ObjectState._clearAllState();
+    if (singleInstance) {
+      SingleInstanceState._clearAllState();
```

Comment:
Unless I miss something, this is the only place that doesn't accept `this.className, this._getStateIdentifier()`. And even this is abstract-able. What prevent us from passing Object (e.g. `this`) to `SingleInstanceState`'s methods?